pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -665,3 +665,69 @@ func (p *Provisioner) setAdminOpsAPIClient() error {
 
 	return nil
 }
+func (p Provisioner) updateAdditionalSettings(ob *bktv1alpha1.ObjectBucket) error {
+	var maxObjectsInt64 int64
+	var maxSizeInt64 int64
+	var err error
+	var quotaEnabled bool
+	maxObjects := MaxObjectQuota(ob.Spec.Endpoint.AdditionalConfigData)
+	maxSize := MaxSizeQuota(ob.Spec.Endpoint.AdditionalConfigData)
```

Comment:
A lot of this method seems like duplicate code from `setAdditionalSettings()` above. Can this be refactored so creation and update call the same helper to process the additional settings?

Reply:
@travisn : I agree that the variables used in the both functions are the same but checks used in both are different.
`setAdditionalSettings` will enable quota if `MaxSize` and `MaxObjects` are set, but `updateAdditionalSettings` checks condition for disabling quota and will enable quota only if existing value changes.

Reply:
Ok I see the differences now. We can keep it separate for now.